src/components/YourCompany/NameTranslations/NameTranslation.vue

```diff
@@ -0,0 +1,314 @@
+import { ActionBindingIF, ConfirmDialogType } from '@/interfaces'
+
+// Mixins
+import { CommonMixin } from '@/mixins'
+import { Action, Getter } from 'vuex-class'
+
+@Component({
+  components: {
+    AddNameTranslation,
+    ListNameTranslation,
+    ConfirmDialog
+  }
+})
+export default class NameTranslation extends Mixins(CommonMixin) {
+  // Refs
+  $refs!: {
+    confirmTranslationDialog: ConfirmDialogType
+  }
+
+  @Prop({ default: () => [] })
+  private nameTranslations!: Array<string>
+
+  @Prop({ default: () => [] })
+  private originalNameTranslations!: Array<string>
+
+  // Properties
+  private drafTranslations: Array<string> = []
+  private isEditing: boolean = false
+  private isAddingNameTranslation = false
+  private editingNameTranslation = ''
+  private editIndex = -1
+
+  private get hasNameTranslationChange (): boolean {
+    return this.drafTranslations.length !== this.originalNameTranslations.length ||
+      !this.drafTranslations.every((value, index) => {
+        return this.originalNameTranslations[index] === value
+      })
+  }
+
+  private setNameTranslations (): void {
+    this.emitNameTranslations(this.drafTranslations)
+    this.emitHaveChanges(this.hasNameTranslationChange)
+    this.isEditing = false
+  }
+
+  private resetNameTranslations (): void {
+    this.drafTranslations = [ ...this.originalNameTranslations ]
+    this.emitNameTranslations(this.drafTranslations)
+    this.emitHaveChanges(false)
+    this.isEditing = false
+  }
+
+  private cancelNameTranslationCorrection () {
+    const nameTranslations = this.nameTranslations || []
+    const hasUnsavedData = this.drafTranslations.length !== nameTranslations.length ||
+      !this.drafTranslations.every((value, index) => {
+        return nameTranslations[index] === value
+      })
+    if (hasUnsavedData) {
+      this.confirmUnsavedChanges()
+    } else {
+      this.isEditing = false
+    }
+  }
+
+  private confirmUnsavedChanges () {
+    // open confirmation dialog and wait for response
+    this.$refs.confirmTranslationDialog.open(
+      'Unsaved Changes',
+      'You have unsaved changes in your Name Translations. Do you want to save these changes?',
+      {
+        width: '45rem',
+        persistent: true,
+        yes: 'Save changes',
+        no: null,
+        cancel: 'Cancel without saving'
+      }
+    ).then(async (confirm) => {
+      if (confirm) {
+        this.setNameTranslations()
+      }
+    }).catch(() => {
+      this.onNameTranslationsPropValueChanged()
+      this.isEditing = false
+    })
+  }
+
+  /** Add or update a name translation
+   *
+   * @param name The name to add
+   */
+  private addName (name: string): void {
+    // Handle name translation adds or updates
+    this.editIndex > -1
+      ? this.drafTranslations[this.editIndex] = name
+      : this.drafTranslations.push(name)
+
+    // Emit haveChanges
+    // this.emitHaveChanges(this.hasNameTranslationChange)
+
+    this.cancelNameTranslation()
```

Comment:
This code doesn't make sense to me (and you have some commented out code)... Why are you canceling the name translation during add?